pkg/network/sriov/sriov.go

```diff
@@ -19,4 +19,77 @@
 
 package sriov
 
+import (
+	"encoding/json"
+	"fmt"
+
+	networkv1 "github.com/k8snetworkplumbingwg/network-attachment-definition-client/pkg/apis/k8s.cni.cncf.io/v1"
+	virtv1 "kubevirt.io/api/core/v1"
+
+	"kubevirt.io/kubevirt/pkg/network/namescheme"
+	"kubevirt.io/kubevirt/pkg/network/vmispec"
+)
+
 const AliasPrefix = "sriov-"
+
+func CreateNetworkPCIAnnotationValue(networks []virtv1.Network, interfaces []virtv1.Interface, networkStatusAnnotationValue string) (string, error) {
+	networkPciMap, err := mapNetworkNameToPCIAddress(networks, interfaces, networkStatusAnnotationValue)
+	if err != nil {
+		return "", err
+	}
+
+	if len(networkPciMap) == 0 {
+		return "", fmt.Errorf("expecting non-emtpy network-pci-map")
+	}
+	networkPciMapBytes, err := json.Marshal(networkPciMap)
+	if err != nil {
+		return "", err
+	}
+
+	return string(networkPciMapBytes), nil
+}
+
+func mapNetworkNameToPCIAddress(networks []virtv1.Network, interfaces []virtv1.Interface, networkStatusAnnotationValue string) (map[string]string, error) {
+	multusInterfaceNameToNetworkStatusMap, err := mapMultusInterfaceNameToNetworkStatus(networkStatusAnnotationValue)
+	if err != nil {
+		return nil, err
+	}
+	networkNameScheme := namescheme.CreateNetworkNameScheme(networks)
+
+	networkPciMap := map[string]string{}
```

Comment:
This map is called `networkPciMap` but it actually maps VMI interface names to PCI addresses.
```suggestion
	ifaceNameToPciAddrMap := map[string]string{}
```

Reply:
it's not. specifically it maps names from vmi.spec.networks[i].Name to PCI-Addresses

Reply:
The loop bellow is iterating over `vmispec.FilterSRIOVInterfaces(interfaces)` and takes their name `sriovIface.Name` which is linking to the network name.
I find this a bit confusing.

Reply:
yes, the interface.name is linking to the network.name.
It's confusing because vmi spec is confusing by design.. 

The reason I am iterating the interfaces is because only interfaces have the "sriov" field.
However the "interface" filed in the multus netowrk status annotation (i.e. net1", "net2"..) is indexed from the network list, and is connected directly to it's order. The link between interfaces and networks is the name (interface.Name points to the network name)..
I'm afraid the confusion is by design..